action.yml

```diff
@@ -6,78 +6,173 @@ inputs:
   db-server-name:
     description: The name of the database server to build the database on.
     required: true
+    default: 'localhost'
   db-server-port:
     description: The port that the database server listens on.
     required: false
     default: '1433'
   db-name:
     description: The name of the database to bubild.
     required: true
-  install-mock-db-objects:
-    description: Specifies whether mock db objects should be used to fill out dependencies. If set to true mock-db-object-nuget-feed-url must also be set, otherwise an error will occur. The expected value is true or false.
-    required: false
+  use-integrated-security:
+    description: Use domain integrated security. If false, a db-username and db-password should be specified. If true, those parameters will be ignored if specified.
+    required: true
     default: 'false'
-  mock-db-object-nuget-feed-url:
-    description: The url to the nuget feed containing the mock database objects. This needs to be set if the install-mock-db-objects flag is set to avoid errors.
+  db-username:
+    description: The username to use to login to the database. This is required if use-integrated-security is false, otherwise it's optional and will be ignored. The default value (sa) references the default system administrator account for SQL Server.
     required: false
-  nuget-username:
-    description: The username for the user to authenticate with the nuget feed. This should be set if install-mock-db-objects is true.
+    default: 'sa'
+  db-password:
+    description: The password for the user logging in to the database. This is required if use-integrated-security is false, otherwise it's optional and will be ignored.
     required: false
-  nuget-password:
-    description: The password for the user to authenticate with the nuget feed. This should be set if install-mock-db-objects is true.
+  migration-files-path:
+    description: The path to the base directory containing the migration files to process with flyway
+    required: true
+  install-mock-db-objects:
+    description: Specifies whether mock db objects should be used to fill out dependencies. If set to true mock-db-object-dependency-list must also be set, otherwise an error will occur. The expected value is true or false.
     required: false
+    default: 'false'
+  mock-db-object-dependency-list:
+    description: A json string containing a list of objects with the name of the dependency package, the version, and the url where the package is stored.
+    required: true
   incremental:
-    description: Specifies whether to drop and recreate the database before building, or apply to the current database. The expected value is true or false.
+    description: Specifies whether to drop and recreate the database before building, or apply to the current database. The expected value is true or false. If true, the create-database-file property will not be used.
```

Comment:
There is only one project that needs this in the CI, but you're right that it could be used to run against a real db server (I don't know if that will ever happen). The one project that needs this does a restore from a backup before running the build so they need to skip the database creation steps.

Reply:
Could this always be true?  We never want to drop and recreate a db.  If it doesn't exist yet, it should build it all?  If so, we could omit this argument.

Reply:
The answer to your question @hpractv is kind of what I responded with initially, but there is a little more I can say. I know the data practices team has run these build steps against persistent (non-build machine) databases before for testing purposes. And they have used the drop/recreate feature on those, from what I understand. So if not for them and their database repo, we could very well remove this flag.